Handle failed refunds

jrodger opened the pull request:
Fixes #312

#### Changes proposed in this Pull Request

* Adds a REST endpoint for receiving webhooks forwarded from WP.com
* Initial action is simply adding an order note when we get an update regarding a failed refund.

#### Local Environment Testing instructions

* Apply server PR https://github.com/Automattic/woocommerce-payments-server/pull/159 to your local server and forward Stripe webhooks to it using the CLI: `./local/bin/listen-to-webhooks.sh`
* Check that the mocked details in `functions.php - wcpay_server_remote_request` match your local setup. This replaces the Jetpack call with a simple HTTP request.
* Apply dev tools PR https://github.com/Automattic/woocommerce-payments-dev-tools/pull/4 to your local site. This allows the simple HTTP request to work without proper Jetpack authentication.
* Make a purchase with test card `4000 0000 0000 5126` ([docs](https://stripe.com/docs/testing#cards))
* Refund the purchase in wp-admin
* The refund will succeed, but then asynchronously fail. Wait for the refund to fail, refresh the page and ensure an order note is added indicating that the refund failed

#### Sandbox Testing instructions

* Apply server PR https://github.com/Automattic/woocommerce-payments-server/pull/159 to your sandbox and forward Stripe webhooks to it using the CLI
* Make a purchase with test card `4000 0000 0000 5126` ([docs](https://stripe.com/docs/testing#cards))
* Refund the purchase in wp-admin
* The refund will succeed, but then asynchronously fail. Wait for the refund to fail, refresh the page and ensure an order note is added indicating that the refund failed

jrodger committed:
Extract DB access from Stripe API client class

This logic is required by other parts of the code base, so extracting
it out allows it to be shared. The unit tests have been left as is for
now, but this also allows for database access to be mocked more easily.

jrodger committed:
Add REST endpoint for handling webhooks from WP.com

The initial implementation just handles refund updated events and adds
an order note in the event of a failure.

vbelolapotkov reviewed (commented):
@jrodger I was able to set up stripe CLI and reproduce testing instruction with webhooks forwarded to the sandbox. Well done!

Some general thoughts:

1. What about tests for webhook controller class?
2. As @RadoslavGeorgiev noticed we have a successful confirmation note first and a failure note afterwards. So it's probably a good idea to update existing messaging for order notes to avoid confusion in case of refund failure. E.g. `Refund request was successfully sent` as an alternative to existing success message.

jrodger committed:
Turn off Squiz.Commenting.FunctionCommentThrowTag.WrongNumber PHPCS rule

This rule is generating some false positives. We can try turning it on
again if it gets fixed in a later PHPCS version. Alternatively we could
start relying on a static code analyser for more robust checks on how
we're using exceptions.

vbelolapotkov reviewed (changes requested):
Great progress @jrodger, thank you ! 🎉 I've tested it both in sandbox mode and with a local server in docker. Changes work as expected. Left some note regarding the changes!

jrodger committed:
Fix whitespace in phpcs.xml.dist

jrodger committed:
Clean up webhook controller test class

jrodger committed:
Tweak naming of read_property function

jrodger committed:
Make exception message translatable

jrodger committed:
Escape HTML in translation string

vbelolapotkov reviewed (approved):
Gave it a final round of tests, works as expected! Thank you @jrodger !
![image](https://user-images.githubusercontent.com/3139099/79959197-83d0ab80-848c-11ea-8d3c-0bdfdffa961a.png)

jrodger committed:
Rename WC_Payments_Rest_Exception to WC_Payments_Rest_Request_Exception
